### ISSUE TITLE
Failing Example: CRD

### ISSUE THREAD

**Issue by jefflantz** [hashicorp/terraform-provider-kubernetes-alpha#71]
### Terraform Version and Provider Version
Terraform v0.12.26
Built provider from source on 6/22/20
### Affected Resource(s)
- CustomResourceDefinition

### Terraform Configuration Files
[exampes/crd/main.tf](examples/crd/main.tf)

### Debug Output

Nothing in debug looks particularly useful, [link regardless](https://gist.github.com/jefflantz/b45d2ade1441d9ce0c8c55086fdcbba7)

### Expected Behavior
Should create a custom resource definition named `testcrdks.hashicorp.com`

### Actual Behavior
Various behaviors I've seen. As written, gives the error
```
rpc error: code = Unknown desc = no matches for apiextensions.k8s.io/v1, Resource=CustomResourceDefinition
```

### Steps to Reproduce
 ```
git clone https://github.com/hashicorp/terraform-provider-kubernetes-alpha.git
cd terraform-provder-kubernetes-alpha
terraform init
terraform apply
```

### Important Factoids
I've tried a few things along the way. 
Setting `server_side_planning = false` gives the error
`rpc error: code = Unknown desc = failed to determine resource GVR: no matches for apiextensions.k8s.io/v1, Resource=CustomResourceDefinition`
Similar error when additionally setting `apiVersion = "v1"` (inspired by [services example](examples/service/service.tf) and [README](README.md))

### References
Don't think it matches any other issues. I had some issues getting to this point but they were fixed by using the latest version.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by mcristina422** [comment]
What version of kubernetes are you using? `v1` CRDs were added in 1.16. Try using the v1beta1 CRDs

**Comment by jefflantz** [comment]
Thanks for the quick response. The output of `kubectl version` is
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.8-gke.15", GitCommit:"9cabee15e0922c3b36724de4866a98f6c2da5e6a", GitTreeState:"clean", BuildDate:"2020-05-01T21:47:04Z", GoVersion:"go1.13.8b4", Compiler:"gc", Platform:"linux/amd64"}
```
Now I get a new error when I run the code, being
```
rpc error: Unknown desc = update dry-run for 'testcrds.hashicorp.com' failed: Apply failed with 4 conflicts: conflicts with "before-first-apply" using apiextensions.k8s.io/v1beta1:
- .spec.group
- .spec.scope
- .spec.names.kind
- .spec.names.plural
```
Could this possibly relate to PR-41?

**Comment by jefflantz** [comment]
Upon further examination, this appears to work on `Minikube`. Not sure why it doesn't work with GCP. Will keep looking.

To make it work, after removing my test clusters from GKE and from my kubeconfig, simply running the following created the crd.
```
minikube start
terraform apply
kubectl get crd
```

Will keep looking into why it isn't working nicely with Google.

**Comment by danawillow** [comment]
I ran into the same issue and resolved it by upgrading my cluster to >=1.17 (as per https://www.hashicorp.com/blog/deploy-any-resource-with-the-new-kubernetes-provider-for-hashicorp-terraform/#current-limitations)

**Comment by jefflantz** [comment]
Ok, to deploy let GKE create a kubernetes 1.17 cluster I had to use the google-beta provider and the google_container_engine_versions data source to get the current rapid channel kubernetes version. That fixed this specific issue, thanks so much! Closed.
